testsuite/formatter.py

```diff
@@ -31,6 +31,10 @@ def update_status_data(self):
             if 'systems' in element:
                 element['systems'] = element['systems']()
 
+        # If the entire feature has been skipped, populate the skip_reason
+        if current_feature['status'] == 'skipped':
```

Comment:
Is this grabbing data from the *final* element of the current feature `elements` array? This is slightly surprising -- does the array in general contain only 1 element if the feature has been skipped? If so, can the skip reason logic live under the conditional here?

Reply:
From what I've seen, if a feature is skipped, the skip_reason is the same and present in all of the elements. At the very least, the last should be more representative than the first, since if a feature is skipped half-way through, the first element would have no reason, and the last would.

That being said, I did it this way because the element variable was conveniently set, not for that justification.